### ISSUE TITLE
Undefined local variable or method `instance' in Rails 3.0.20.2

### ISSUE THREAD

**Issue by seeflanigan** [bkeepers/dotenv#151]
Hi @bkeepers,

Thanks for sharing yet another awesome thing with the community!

I'm attempting to use `dotenv` in a Rails 3.0.2 application by requiring `dotenv-rails` via `Gemfile`.

However, on initialization, the `#self.load` method seems to hit a snag when trying to call `#instance.load`, because instance is undefined in that scope. 

Here's the error:

    /usr/local/lib/ruby/gems/2.1.0/gems/dotenv-rails-1.0.2/lib/dotenv/rails.rb:27:in `load': undefined local variable or method `instance' for Dotenv::Railtie:Class (NameError)
        from /usr/local/lib/ruby/gems/2.1.0/gems/dotenv-rails-1.0.2/lib/dotenv/rails.rb:10:in `block in <class:Railtie>'

The application also uses `foreman (0.6.3)` which requires `dotenv (>= 0.7)`, both are old versions, so I'm not sure if this is causing a conflict but the error occurs regardless of doing `bundle exec foreman start` or `bundle exec rails s`, and it comes from the same version `1.0.2` of `dotenv-rails`.

Is this a Rails 3 vs Rails 4 issue? (Where `instance` is available in Rails 4, but not 3?) I'm looking into upgrading this application anyway so it may soon be moot, but I would like to better understand what's happening here.

Any enlightenment you can help me find here will be highly appreciated, and thanks again for sharing the gifts of yourself and your time with this here software community!

Best,
Cory

**Comment by keithlayne** [comment]
At some point after Rails 3.0, `instance` was added to `Railtie::Configurable`.  It's super-simple:

```ruby
def instance
  @instance ||= new
end
```

I'm stuck with a 3.0 app for now.  I got this to work by adding the following in application.rb after the `Bundler.require` line and before the app module is defined:

```ruby
module Rails
  class Railtie
    def self.instance
      @instance ||= new
    end
  end
end
```

I know this is really ugly, but I did it because it seems like initializers get loaded too late to be used for this.  This at least let the app start using dotenv.  I'll report back if I discover any other issues with this workaround.
